django/db/models/fields/json.py

```diff
@@ -385,10 +385,16 @@ def as_sqlite(self, compiler, connection):
         datatype_values = ",".join(
             [repr(datatype) for datatype in connection.ops.jsonfield_datatype_values]
         )
-        return (
-            "(CASE WHEN JSON_TYPE(%s, %%s) IN (%s) "
-            "THEN JSON_TYPE(%s, %%s) ELSE JSON_EXTRACT(%s, %%s) END)"
-        ) % (lhs, datatype_values, lhs, lhs), (tuple(params) + (json_path,)) * 3
+        if connection.get_database_version() >= (3, 38):
+            return (
+                "(CASE WHEN JSON_TYPE(%s, %%s) IN (%s) "
+                "THEN JSON_TYPE(%s, %%s) ELSE %s ->> %%s END)"
+            ) % (lhs, datatype_values, lhs, lhs), (tuple(params) + (json_path,)) * 3
+        else:
+            return (
+                "(CASE WHEN JSON_TYPE(%s, %%s) IN (%s) "
+                "THEN JSON_TYPE(%s, %%s) ELSE JSON_EXTRACT(%s, %%s) END)"
+            ) % (lhs, datatype_values, lhs, lhs), (tuple(params) + (json_path,)) * 3
```

Comment:
What about comment in SQLite operations:

https://github.com/django/django/blob/cf03aa4e94625971852a09e869f7ee7c328b573f/django/db/backends/sqlite3/operations.py#L27-L29

Do we still need a `JSON_TYPE` workaround? If yes, I don't see much value in this change, TBH :shrug:

Reply:
From what I can tell the `->>` operator behaves the same way as `JSON_EXTRACT` so we still need the `JSON_TYPE` workaround if we use it.

The `->` operator, handles the null, false, true values the way as the CASE statement with JSON_TYPE:

```
sqlite> select JSON_TYPE('null', '$');
null
sqlite> select JSON_TYPE('true', '$');
true
sqlite> select JSON_TYPE('false', '$');
false

sqlite> select 'null' -> "$";
null
sqlite> select 'true' -> "$";
true
sqlite> select 'false' -> "$";
false
```

But there are differences, one example:

```
sqlite> select JSON_EXTRACT('"7"', "$");
7
sqlite> select '"7"' -> "$";
"7"
```

This causes `failures=29, errors=2` in test_jsonfield.py, and I'm not exactly sure how to fix it.

I think you might be correct about there not being much value in having this merged. I'm happy if we can close the ticket as won't do instead.